### PR TITLE
Added few missing functionalities in NN and tensors.

### PR DESCRIPTION
**Pull request by interesaaat**
Also testing the pipelines

### CODE REVIEW

File: src/TorchSharp/NN/Linear.cs [hunk 1]
```diff
@@ -10,47 +10,43 @@ public Linear(IntPtr handle) : base(handle)
         {
         }
 
-        [DllImport("LibTorchSharp")]
-        private static extern IntPtr THSNN_linearModule(long input_size, long output_size, bool with_bias);
+        [DllImport("libTorchSharp")]
```
Comment (artidoro) [inline src/TorchSharp/NN/Linear.cs]: The errors you get for Linux are probably due to the lower case of libTorchSharp.
Linux is case sensitive, and I think I used LibTorchSharp everywhere else in the code.

Let me know if that does not work.

**Comment by joemoorhouse** [comment]
Oh great; was going to ask if someone was working on scalar-tensor ops! Is idea to have implicit conversions to Scalar or add overloads? e.g. public static implicit operator Scalar(double value){…}

File: src/TorchSharp/NN/Linear.cs (hunk 1 above)
Comment (interesaaat) [reply src/TorchSharp/NN/Linear.cs]: Ah-ha! Let me fix this

**Comment by interesaaat** [comment]
@joemoorhouse for the moment we you need to explicitly convert into scalars. For example we do 
```cs
 weights.Eq(1.ToScalar()).Sum(new[] { 1L }) * (-1).ToScalar() + 0.5.ToScalar();
```
which is terrible! Please open an issue and feel free to fix this if you want to!

**Comment by interesaaat** [comment]
Tests pass now (thanks @artidoro). I am going to merge this.

**Pull request merged**